src/Forms/CommentForm.php

```diff
@@ -132,21 +131,28 @@ public function __construct($name, CommentingController $controller)
         // Set it so the user gets redirected back down to the form upon form fail
         $this->setRedirectToFormOnValidationError(true);
 
-        // load any data from the cookies
-        if ($data = Cookie::get('CommentsForm_UserData')) {
-            $data = Convert::json2array($data);
+        // load any data from the session
+        $data = $this->getSessionData();
+        if (is_array($data)) {
```

Comment:
I think it's a bit nicer to do `if (!is_array($data)) { return; }` to skip the next set of lines being indented